library/core/tests/slice.rs

```diff
@@ -1356,6 +1359,22 @@ mod slice_index {
             bad: data[7..=6];
             message: "out of range";
         }
+
+        in mod boundpair_len {
+            data: [0, 1, 2, 3, 4, 5];
+
+            good: data[(Bound::Included(6), Bound::Unbounded)] == [];
+            good: data[(Bound::Unbounded, Bound::Included(5))] == [0, 1, 2, 3, 4, 5];
+            good: data[(Bound::Unbounded, Bound::Excluded(6))] == [0, 1, 2, 3, 4, 5];
+            good: data[(Bound::Included(0), Bound::Included(5))] == [0, 1, 2, 3, 4, 5];
+            good: data[(Bound::Included(0), Bound::Excluded(6))] == [0, 1, 2, 3, 4, 5];
+            good: data[(Bound::Excluded(5), Bound::Excluded(6))] == [];
```

Comment:
Let's add a few more test cases here that ensure that combinations of ranges within the data itself return useful results, so for instance:

- `(Bound::Included(2), Bound::Excluded(4))`
- `(Bound::Excluded(1), Bound::Included(4))`

Reply:
Done